src/core/build_target.go

```diff
@@ -1190,22 +1199,22 @@ func (target *BuildTarget) AddTool(tool BuildInput) {
 
 // AddTestTool adds a new test tool to the target.
 func (target *BuildTarget) AddTestTool(tool BuildInput) {
-	target.testTools = append(target.testTools, tool)
+	target.Test.tools = append(target.Test.tools, tool)
```

Comment:
Is it guaranteed that `target.Test` is non nil at this stage?

Reply:
Same for the other functions below

Reply:
There's only one* place where we should be constructing targets: `src/parse/asp.createTarget(...)`. When we construct the target we guard against calling these functions with an `if target.IsTest() {}` so it should be set. If it is a target, we make sure this field is initialised. 

That being said, don't look at my PR. I'm infallible and there's nothing to see there. ;P

* okay there are a couple others but they're not taking in user input so are safe.